src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -288,6 +313,54 @@ private class Emit(
             srvb.advance(),
             i := i + 1),
           srvb.offset))
+      case x@ArrayFilter(a, name, body) =>
+        val t = x.typ
+        val keept = TArray(TBoolean())
+        val srvbkeep = new StagedRegionValueBuilder(fb, keept)
+        val srvb = new StagedRegionValueBuilder(fb, t)
+        val etiin = coerce[Any](typeToTypeInfo(t.elementType))
+        val xa = fb.newLocal[Long]("af_a")
+        val xmv = mb.newBit()
+        val xvv = fb.newLocal(name)(etiin)
+        val i = fb.newLocal[Int]("af_i")
```

Comment:
I think it'd be clearer to move these four lines down and/or use newlines to group related lines.